internal/services/reservations_service.go

```diff
@@ -58,6 +58,7 @@ func ListReservations(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetReservationDetail(w http.ResponseWriter, r *http.Request) {
+	provider := chi.URLParam(r, "TYPE")
```

Comment:
Hmm okay I remember deleting this :-)

Can we check if the provider passed in as URL parameter matches provider in the database? If not, we should definitely print out an error. Tho I think we won't be able to map fields form the database correctly before we actually load up the type from SQL column. So this might only be post check, hmm.